pageserver/src/tenant/timeline.rs

```diff
@@ -1466,10 +1466,14 @@ impl Timeline {
             return Ok(None);
         };
 
-        match local_layer.evict_and_wait().await {
+        // curl has this by default
+        let timeout = std::time::Duration::from_secs(120);
+
+        match local_layer.evict_and_wait(timeout).await {
```

Comment:
We know it's cancel-safe, let's just use Duration::MAX or whatever it is, and I suppose this method and `evict_and_wait` should take a CancellationToken.

I can see us use this API in an emergency Python script which doesn't want any timeout.

If you disagree with that, at least have a better comment that sets context why `curl` is important .


```suggestion
        // mgmt APIs like this one are typically called from curl, which has a 120 timeout by default
        let timeout = std::time::Duration::from_secs(120);

        match local_layer.evict_and_wait(timeout).await {
```

Reply:
I will reconsider this when perhaps making the timeout external to `evict_and_wait` once again in the future.